bumble/smp.py

```diff
@@ -975,9 +967,20 @@ def display_or_input_passkey(
         self, next_steps: Optional[Callable[[], None]] = None
     ) -> None:
         if self.passkey_display:
-            self.display_passkey()
-            if next_steps is not None:
-                next_steps()
+
+            async def display_passkey():
+                await self.display_passkey()
+                if next_steps is not None:
+                    next_steps()
+
+            try:
+                utils.cancel_on_event(
+                    self.connection,
+                    'disconnection',
```

Comment:
```suggestion
                    self.connection.EVENT_DISCONNECTION,
```

Reply:
I ended up creating a help function for the common call: `utils.cancel_on_event(self.connection, 'disconnection', foo()` --> `self.connection.cancel_on_disconnection(foo())`